File: Domain/Post.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain
{
    public class Post
    {
        public Guid Id { get; set; }
        public string Slug { get; set; }
        public virtual AppUser Author { get; set; }
        public virtual Category Category { get; set; }
        public string Image { get; set; }
        public string Title { get; set; }
        public string SubTitle { get; set; }
        public string Content { get; set; }
        public virtual ICollection<Comment> Comments { get; set; }
        public virtual ICollection<Reaction> Reactions { get; set; }
        public DateTime PublishDate { get; set; }
    }
}

File: client-app/src/app/models/post.ts
import { ICategory } from "./category";
import { IAuthor } from "./author";
import { IComment } from "./comment";
import { IReaction } from "./reaction";

export interface IPost {
  id: string;
  slug: string;
  image: string;
  title: string;
  subTitle: string;
  hasLiked: boolean;
  content: string;
  positiveReactionsCount: number;
  timeToRead: string;
  publishDate: Date;
  category: ICategory;
  author: IAuthor;
  comments: IComment[];
  reactions: IReaction[];
  recommendedPosts: IPostRecommend[];
}

export interface IPostRecommend {
  id: string;
  slug: string;
  image: string;
  title: string;
  content: string;
  publishDate: Date;
  category: ICategory;
}

export interface IPostsEnvelope {
  posts: IPost[];
  postsCount: number;
}

export interface IPostsForm {
    id?: string;
    slug: string;
    title: string;
    subTitle: string;
    image: Blob | string;
    content: string;
    categoryCode: string;
}

export class IPostFormValues implements IPostsForm {
    slug = "";
    title = "";
    subTitle = "";
    image = new Blob();
    content = "";
    categoryCode = "";
}
